edvaudin.UnitTests/CodingTrackerTests.cs

```diff
@@ -0,0 +1,122 @@
+using CodingTracker;
+
+namespace edvaudin.UnitTests
+{
+    [TestClass]
+    public class CodingTrackerTests
+    {
+        [TestMethod]
+        [DataRow("beans")]
+        [DataRow("")]
+        [DataRow("x")]
+        [DataRow("2")]
+        public void IsValidFilterOption_WithInvalidInput_ReturnsFalse(string option)
+        {
+            bool result = Validator.IsValidFilterOption(option);
+            Assert.IsFalse(result);
+        }
+
+        [TestMethod]
+        [DataRow("a")]
+        [DataRow("d")]
+        [DataRow("w")]
+        [DataRow("m")]
+        [DataRow("y")]
+        public void IsValidFilterOption_WithValidInput_ReturnsTrue(string option)
+        {
+            bool result = Validator.IsValidFilterOption(option);
+            Assert.IsTrue(result);
+        }
+
+        [TestMethod]
+        [DataRow("beans")]
+        [DataRow("")]
+        [DataRow("x")]
+        [DataRow("2")]
+        public void IsValidOption_WithInvalidInput_ReturnsFalse(string option)
+        {
+            bool result = Validator.IsValidOption(option);
+            Assert.IsFalse(result);
+        }
+
+        [TestMethod]
+        [DataRow("v")]
+        [DataRow("a")]
+        [DataRow("d")]
+        [DataRow("u")]
+        [DataRow("srt")]
+        [DataRow("stp")]
+        [DataRow("0")]
+        public void IsValidOption_WithValidInput_ReturnsTrue(string option)
+        {
+            bool result = Validator.IsValidOption(option);
+            Assert.IsTrue(result);
+        }
+
+        [TestMethod]
+        [DataRow("")]
+        [DataRow(" ")]
+        [DataRow("40-03-24 23-03-22")]
+        [DataRow("12-03-24 25-03-22")]
+        [DataRow("12:03:24 04:03:22")]
+        [DataRow("12/03/24 04:03:22")]
+        public void IsValidDateInput_WithInvalidInput_ReturnsFalse(string input)
+        {
+            bool result = Validator.IsValidDateInput(input);
+            Assert.IsFalse(result);
+        }
+
+        [TestMethod]
+        [DataRow("05-02-23 22-31-00")]
+        [DataRow("17-01-23 05-24-20")]
+        [DataRow("17-12-01 04-24-20")]
+        public void IsValidDateInput_WithValidInput_ReturnsTrue(string input)
+        {
+            bool result = Validator.IsValidDateInput(input);
```

Comment:
I'm getting this error: 

![image](https://user-images.githubusercontent.com/108711476/217976789-d60134c8-e307-4c25-a6e7-00e5b99e41a6.png)